core/models.py

```diff
@@ -106,9 +114,23 @@ def articles(self):
                 default=models.Value(0),
                 output_field=models.IntegerField())).order_by("-sticky", "-first_published_at")
 
-        return self.get_article_set(self.number_of_columns_of_articles,
-                                    self.number_of_rows_of_articles, articles,
-                                    [])
+        self._articles = self.get_article_set(self.number_of_columns_of_articles,
+                                              self.number_of_rows_of_articles, articles,
+                                              [])
+
+        return self._articles
+
+    @property
+    def most_popular_article(self):
+        if self._most_popular_article is not None:
+            return self._most_popular_article
+
+        # flatten the list of lists with generator
+        articles = itertools.chain.from_iterable(self.articles)
+        articles_by_popularity = sorted([(page.analytics.last_week_views, page) for page in articles])
```

Comment:
could probably do something like the following to avoid having the extra rank variable that isn't used, but probably not critical:
articles_by_popularity = sorted(articles, key=lambda x: x.analytics.last_week_views)